openedx/core/djangoapps/schedules/config.py

```diff
@@ -9,4 +14,28 @@
     flag_undefined_default=False
 )
 
-DEBUG_MESSAGE_WAFFLE_FLAG = WaffleFlag(WAFFLE_FLAG_NAMESPACE, u'enable_debugging')
+DEBUG_NUDGE_MESSAGE_SAMPLE_NAME = u'schedules.enable_nudge_message_debugging'
+DEBUG_NUDGE = u'schedules.enable_nudge_debugging'
+
+
+@contextmanager
+def enable_debugging_if_necessary():
+    existing_level = None
+    if waffle.switch_is_active(DEBUG_NUDGE) or waffle.sample_is_active(DEBUG_NUDGE_MESSAGE_SAMPLE_NAME):
+        logger = logging.getLogger()
+        existing_level = logger.getEffectiveLevel()
+        logger.setLevel(logging.DEBUG)
+        for handler in logger.handlers:
+            if handler.name == u'local':
+                handler.setLevel(logging.DEBUG)
+                break
+    try:
+        yield
+    finally:
+        if existing_level is not None:
```

Comment:
Wouldn't we want to reset the logging level back to `None` if there was no preexisting value?

Reply:
ou - good catch, not sure if getEffectiveLevel can return None or not.